### PR TITLE
Add function to parse quantity

### LINKED ISSUE #36: Add means to compare quantities
## Background

K8s converts user input [quantities](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) to "canonical form":

> Before serializing, Quantity will be put in "canonical form". This means that Exponent/suffix will be adjusted up or down (with a corresponding increase or decrease in Mantissa) such that: a. No precision is lost b. No fractional digits will be emitted c. The exponent (or suffix) is as large as possible. The sign will be omitted unless the number is negative.
>
> Examples: 1.5 will be serialized as "1500m" 1.5Gi will be serialized as "1536Mi"

Additional examples:

| User input                       | K8s representation            |
|----------------------------------|-------------------------------|
| `{"memory": "0.9Gi"}`            | `{"memory": "966367641600m"}` |
| `{"cpu": "0.30000000000000004"}` | `{"cpu": "301m"}`             |

## Use case

When patching a statefulset's resource limits, it would be handy to be able to compare a quantity's setpoint 

```
statefulset.spec.template.spec.containers[i].resources
```

to the active podspec

```
pod.spec.containers[i].resources
```

Currently, two equivalent `ResourceRequirements` instances could differ under `==` comparison and require manual interconversion.

## Additional thoughts
- k8s-client has a [`parse_quantity`](https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py) function, but having both lightkube and kubernetes deps is not ideal.
- [bitmath](https://pypi.org/project/bitmath/) doesn't support millis and is awkward to "abuse" for cpu values.
- Implementing [comparisons](https://peps.python.org/pep-0207/) for `class ResourceRequirements` might be ackward.

### PR DESCRIPTION
**Pull request by sed-i**
This PR add a utility function for parsing K8s "quantity" from string.
The value is parsed into an absolute (unitless) [`decimal.Decimal`](https://docs.python.org/3/library/decimal.html) instance, and not into the K8s "canonical representation". This way the code is simpler and values are readily comparable.

References:
1. https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py
2. https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go

Fixes #36.

### CODE REVIEW

**Review by gtsystem — COMMENTED** [review]
Hi, thanks for your PR. Can you add this function to the documentation together with an example? If you introduce the `utils` sub-package you can create `docs/utils.md`

**Comment by sed-i** [comment]
Thanks @gtsystem! Ready for re-review.

File: lightkube/utils/quantity.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    * **second** `ResourceRequirements` or `dict` - The second item to compare.
+
+    **returns**  True, if both arguments are numerically equal; False otherwise.
+    """
+    def is_eq(first_dict: Optional[dict], second_dict: Optional[dict]) -> bool:
+        if first_dict == second_dict:
+            # This covers two cases: (1) both args are None; (2) both args are identical dicts.
+            return True
+        if first_dict and second_dict:
+            ks = ("cpu", "memory")
```
[the remark this answers is not in the document]
Comment (sed-i) [reply lightkube/utils/quantity.py:109]: Fixed in next commit.

File: lightkube/utils/quantity.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        raise ValueError("Invalid numerical value") from e
+
+
+@overload
+def equals_canonically(first: ResourceRequirements, second: ResourceRequirements) -> bool:
+    ...
+
+
+@overload
+def equals_canonically(first: Optional[dict], second: Optional[dict]) -> bool:
```
[the remark this answers is not in the document]
Comment (gtsystem) [reply lightkube/utils/quantity.py:109]: Got it, I was confused by the function description "Compare two resource requirements for numerical equality." I assumed that the only acceptable type is `ResourceRequirements` or the equivalent dict. Maybe we can clarify further?

No need to introduce the TypedDict as anyway all the keys are optional and we can have further keys.

[the remark this answers is not in the document]
Comment (sed-i) [reply lightkube/utils/quantity.py:109]: Fixed in next commit.

File: lightkube/core/internal_models.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import sys
 
 try:
-    from ..models import meta_v1, autoscaling_v1
+    from ..models import meta_v1, autoscaling_v1, core_v1
```
[the remark this answers is not in the document]
Comment (sed-i) [reply lightkube/core/internal_models.py:4]: Fixed in next commit.

File: lightkube/utils/quantity.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    True
+
+    **Parameters**
+
+    * **first** `ResourceRequirements` or `dict` - The first item to compare.
+    * **second** `ResourceRequirements` or `dict` - The second item to compare.
+
+    **returns**  True, if both arguments are numerically equal; False otherwise.
+    """
+    if isinstance(first, Optional[dict]) and isinstance(second, Optional[dict]):
```
Comment (gtsystem) [inline lightkube/utils/quantity.py:132]: Runtime error as typing constructs (`Optional`) cannot be used here.

Comment (sed-i) [reply lightkube/utils/quantity.py:132]: Didn't show up when running tests locally with `platform linux -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0`.
Fixed in next commit.

**Review by gtsystem — APPROVED** [review]

**Comment by gtsystem** [comment]
Approved and merged. We have another contribution in progress, once that is completed I will create a public release. Thanks again.

**Pull request merged**